frame/pallet-xcm-transactor/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+// This file is part of Astar.
+
+// Copyright (C) 2019-2023 Stake Technologies Pte.Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// Astar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Astar is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Astar. If not, see <http://www.gnu.org/licenses/>.
+
+//! Pallet to handle XCM Callbacks.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{pallet_prelude::*, PalletId};
+use frame_system::pallet_prelude::*;
+pub use pallet::*;
+use pallet_contracts::Pallet as PalletContracts;
+use pallet_xcm::Pallet as PalletXcm;
+use sp_core::H160;
+use sp_runtime::traits::{AccountIdConversion, Zero};
+use sp_std::prelude::*;
+use xcm::prelude::*;
+
+pub mod chain_extension;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_system::Config as SysConfig;
+    use pallet_xcm::ensure_response;
+    pub use xcm_ce_primitives::{QueryConfig, QueryType};
+
+    /// Response info
+    #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+    pub struct ResponseInfo<AccountId> {
+        pub query_id: QueryId,
+        pub query_type: QueryType<AccountId>,
+        pub response: Response,
+    }
+
+    /// Query infor
+    #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+    pub struct QueryInfo<AccountId> {
+        pub query_type: QueryType<AccountId>,
+        pub querier: Junctions,
+    }
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_xcm::Config + pallet_contracts::Config {
```

Comment:
That is tight coupling with `pallet_contracts`. Fine for PoC, but for production, it's better to handle such requirements via associated types.